Khan/perseus
# Implement the "angles" snapTo type for Polygon

## Summary:
Polygon previously was only able to snap to grid lines. Functionality was needed to allow it to snap to whole angles regardless of the grid or snapStep. This implements that feature.
 
Issue: LEMS-1855

## Test plan:
- Confirm all tests pass
- Confirm test coverage for new code is good
- Run `yarn dev` in the command line
- Navigate to localhost:5173
- Turn on mafs flags for all graphs
- Confirm the new gallery test for snapTo angles works correctly
- Confirm the other types of graphs work correctly
- Go to the flipbook and enter the following content for polygon
```
{"content":"[[☃ interactive-graph 1]]\n","images":{"web+graphie://ka-perseus-graphie.s3.amazonaws.com/4fbb5026607ec9ee083b365aa6bcda133adca567":{"width":440,"height":425}},"widgets":{"interactive-graph 1":{"type":"interactive-graph","alignment":"default","static":false,"graded":true,"options":{"step":[2,2],"backgroundImage":{"url":null},"markings":"graph","labels":["x","y"],"showProtractor":false,"showRuler":false,"showTooltips":false,"rulerLabel":"","rulerTicks":10,"range":[[-10,10],[-10,10]],"gridStep":[1,1],"snapStep":[1,1],"graph":{"type":"polygon","numSides":4,"showAngles":true,"snapTo":"grid"},"correct":{"type":"polygon","numSides":4,"coords":[[4,3],[4,7],[-1,7],[-1,3]],"snapTo":"grid"}},"version":{"major":0,"minor":0}}}}
```
- Move the points in "snapTo" grid mode and confirm the points snap to the grid
- Move the polygon and confirm it snaps to the grid
- Change the "snapTo" property of the graph object in the content to `angles`
- Move the points and confirm the angles are whole numbers and do not snap to the grid
- Move the entire polygon and confirm it does not snap to the grid and the angles stay whole numbers
